ichnaea/conftest.py

```diff
@@ -29,6 +28,11 @@
 from ichnaea.taskapp.config import init_worker, shutdown_worker as shutdown_celery
 from ichnaea.webapp.config import main, shutdown_worker as shutdown_app
 
+# Enable pytest assertion rewriting
+pytest.register_assert_rewrite("markus.testing")
+from markus.testing import MetricsMock  # noqa: E402
```

Comment:
This was needed to get pytest rich asserts. If I understand the pytest docs on [assertion rewriting](https://docs.pytest.org/en/latest/writing_plugins.html#assertion-rewriting), this could be done in ``markus.__init__.py``, with some magic to keep pytest from being required.



Before:

```
Traceback (most recent call last):
  File "/app/ichnaea/content/tests/test_views.py", line 112, in test_content
    metricsmock.assert_incr_once('request', tags=[f"path:{metric_path}", "method:get", f"status:{status}"])
  File "/usr/local/lib/python3.8/site-packages/markus/testing.py", line 106, in assert_incr_once
    assert len(self.filter_records(INCR, stat=stat, value=value, tags=tags)) == 1
AssertionError
```

After:
```
Traceback (most recent call last):
  File "/app/ichnaea/content/tests/test_views.py", line 117, in test_content
    metricsmock.assert_timing_once('request.timing', tags=tags)
  File "/usr/local/lib/python3.8/site-packages/markus/testing.py", line 130, in assert_timing_once
    assert len(self.filter_records(TIMING, stat=stat, value=value, tags=tags)) == 1
AssertionError: assert 0 == 1
 +  where 0 = len([])
 +    where [] = <bound method MetricsMock.filter_records of <markus.testing.MetricsMock object at 0x7fa8e097d040>>('timing', stat='request.timing', value=None, tags=['path:/api', 'method:get'])
 +      where <bound method MetricsMock.filter_records of <markus.testing.MetricsMock object at 0x7fa8e097d040>> = <markus.testing.MetricsMock object at 0x7fa8e097d040>.filter_records
```

It's an improvement to be able to see the tags after string formatting. The next step for me was always to run in ``--pdb`` mode and peek at ``self.records``. It would be lovely if this was part of the assertion failure, but I think it would take some deep pytest magic.

Reply:
I've totally hit that issue, but hadn't looked into it. If you can write up an issue in Markus for this, that'd be really great because we should fix it.

Also, when I hit issues with markus-related testing, I do this:

```
metricsmock.print_records()
```
and they get dumped to stdout which is easy to see in pytest output.

Reply:
Opened https://github.com/willkg/markus/issues/73 for assertion rewriting (fixable), and https://github.com/willkg/markus/issues/74 for including records in test failures (wishlist, may be impossible)